cmake/FindLIBHWLOC.cmake

```diff
@@ -4,6 +4,33 @@
 
 message(STATUS "Checking for module 'libhwloc' using find_library()")
 
+# When user choose to 
+if (UMF_LINK_HWLOC_STATICALLY)
+    message("Add hwloc as a external project")
+    include(ExternalProject)
+
+    set(HWLOC_URL "https://github.com/open-mpi/hwloc/archive/refs/heads/master.zip")
+    ExternalProject_Add(hwloc
```

Comment:
why this way instead of doing this similar to level-zero? 
we could have something like:

```
if (UMF_LINK_HWLOC_STATICALLY) 
    include(FetchContent)
    set(HWLOC_REPO "....git")
    set(HWLOC_TAG ...)

    FetchContent_Declare(..)
    ...
else()
    pkg_check_modules(LIBHWLOC hwloc>=2.3.0)
    if(NOT LIBHWLOC_FOUND)
        find_package(LIBHWLOC 2.3.0 REQUIRED hwloc)
    endif()
    # add PATH to DLL on Windows
    set(DLL_PATH_LIST
        "${DLL_PATH_LIST};PATH=path_list_append:${LIBHWLOC_LIBRARY_DIRS}/../bin")
endif()
```

and IMO this code should be added to CMakeLists.txt, not FindLIBHWLOC.cmake